cawg_identity/examples/cawg.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+//! Example App that generates a CAWG manifest for a given file
+//! and validates the identity assertion.
+//!
+//! This example is not supported on WASI targets.
+//!
+//! ```bash
+//! cargo run --example cawg -- /path/to/source/file /path/to/destination/file
+//! ```
+mod cawg {
+    use std::path::Path;
+
+    use anyhow::Result;
+    use c2pa::{AsyncSigner, Builder, Reader, SigningAlg};
+    use c2pa_crypto::raw_signature;
+    use cawg_identity::{
+        builder::{AsyncIdentityAssertionBuilder, AsyncIdentityAssertionSigner},
+        validator::CawgValidator,
+        x509::X509CredentialHolder,
+    };
+    use serde_json::json;
+
+    const CERTS: &[u8] = include_bytes!("../../sdk/tests/fixtures/certs/ed25519.pub");
+    const PRIVATE_KEY: &[u8] = include_bytes!("../../sdk/tests/fixtures/certs/ed25519.pem");
+
+    fn manifest_def() -> String {
+        json!({
+            "claim_generator_info": [
+                {
+                    "name": "c2pa cawg test",
+                    "version": env!("CARGO_PKG_VERSION")
+                }
+            ],
+            "assertions": [
+                {
+                    "label": "c2pa.actions",
+                    "data": {
+                        "actions": [
+                            {
+                                "action": "c2pa.opened",
+                            }
+                        ]
+                    }
+                },
+                {
+                    "label": "cawg.training-mining",
+                    "data": {
+                    "entries": {
+                        "cawg.ai_inference": {
+                        "use": "notAllowed"
+                        },
+                        "cawg.ai_generative_training": {
+                        "use": "notAllowed"
+                        }
+                    }
+                    }
+                }
+            ]
+        })
+        .to_string()
+    }
+
+    /// Creates a CAWG signer from a certificate chains and private keys.
+    fn async_cawg_signer() -> Result<impl AsyncSigner> {
+        let c2pa_raw_signer = raw_signature::async_signer_from_cert_chain_and_private_key(
+            CERTS,
+            PRIVATE_KEY,
+            SigningAlg::Ed25519,
+            None,
+        )?;
+
+        let cawg_raw_signer = raw_signature::async_signer_from_cert_chain_and_private_key(
```

Comment:
I'd recommend using different certs for the two cases so we can demonstrate that the right cert goes in the right place.

Reply:
I'll do some more work on this later. I'd like to have a method that accepts two raw signers.